### ISSUE TITLE
Code Editor extension code clashing with external widgets code

### ISSUE THREAD

**Issue by singharpit94** [holoviz/panel#6693]
I am using panel 1.3.8 and not able to use any react based ipywidgets (via ipywidgets-bokeh ) which makes use of **window.require** and **window.define** method along with codeeditor extension.

The primary reason for that are [pre-requirejs](https://github.com/holoviz/panel/blob/main/panel/dist/pre_require.js) and [post-requirejs](https://github.com/holoviz/panel/blob/main/panel/dist/pre_require.js) files where we are overriding the default require and define methods with custom ones.

When we override, the context of both require and define gets affected and we are not able to use methods attach to require context such as `require.config`. Even if we attempt to modify our usage with the new variables  (` _require_requirejs` or `_define_requirejs`) which are being assigned the original values, the resolution of the required modules are not correct and I bump into several new issues ,one of which says
![image](https://github.com/holoviz/panel/assets/7591673/1d6de4e2-8663-4489-b754-0887b662d2e9)

In my local setup, when I comment these overrides my widgets always load successfully though there are some issues with loading the ace editor.

Unfortunately our internal setup is complex and sharing a reproducer is not straight forward but in general I believe we should refrain from overriding these global methods which have associated context variables to avoid usages of them in other libraries.

Can you please check this and suggest way ahead?

**Comment by singharpit94** [comment]
Hi @ndmlny-qs , I actually did not follow if the above changes are intended to be tried at the application/extension layer or in the panel code itself.  Seems like to be in panel code but correct me if I am wrong.

**Comment by ndmlny-qs** [comment]
I do not think we need to make changes just yet, as I would like to see if the check happening on lines 723 and 726 above are picking up JS files in your app that it was not intended to.

I would like to see if when you put a breakpoint below line 722 above that you get `True` back with the `any('ace' in jsf for jsf in js_files)` check. If this is the case, and you do not have the `ace` editor as a requirement in your app, then we need to update the check since we are getting a false positive back from it.

**Comment by singharpit94** [comment]
Correct me if I am wrong but I think I will have the ace requirement in my app if I am using the **'codeeditor'** extension. Below is an example code

```
import panel as pn

pn.extension('codeeditor')

def simple():
    

    py_code = "import sys"
    editor = pn.widgets.CodeEditor(value=py_code, sizing_mode='stretch_width', language='python', height=300)

    # Create a button
    button = pn.widgets.Button(name='Dummy Button', button_type='primary')

    # Return the layout
    return pn.Row(react_widget, button, editor)

simple().servable()
```

I actually tried few more things and found that if I avoid popping out the requirejs files from this [line](https://github.com/holoviz/panel/blob/bfbba8e29facd5b26fd499192e41c46e4e3fbe6f/panel/io/resources.py#L722), it actually loads my react widget and code editor fine, though I will need to test it more to cover all cases.

Any thoughts ?

**Comment by philippjfr** [comment]
Resolved by using a version of Ace editor that doesn't clobber require.js
